### PR TITLE
Suit refactor, fixes and general cleanup

### LINKED ISSUE #9844: You can't unzip jackets, unbutton coats, etc. while buckled
I'm not exactly sure just how deep this rabbit hole goes. It works just fine with hoodies, but not with peacoats or department jackets (at least). I haven't tested anything beyond them but it seems pretty odd.

### PR DESCRIPTION
**Pull request by alsoandanswer**
Fixes #9844 (was due to malignant if() check)

Makes it so suit buttoning/unbuttoning code is less shit.
also makes it take up less code.

Also removes some unused suits and stuff.
namely, technicolor detective stuff (looked bad, had ancient shitcode attached which i am not bothering with)
and the other stuff weren't really referenced anywhere else in the code so w/e

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

File: code/modules/clothing/suits/jobs.dm [hunk 1]
```diff
@@ -213,40 +177,33 @@ obj/item/clothing/suit/apron/overalls/blue
...
-		item_state = icon_open
-		icon_state = icon_open
-		to_chat(usr, SPAN_NOTICE("You unzip the hazard vest."))
-	else
-		to_chat(usr, SPAN_NOTICE("You attempt to zip up your [src], before promptly realising how silly you are."))
-		return
+		return 0
+
+	opened = !opened
+	to_chat(usr, "You [opened ? "unzip" : "zip"] \the [src].")
```
Comment (NonQueueingMatt) [inline code/modules/clothing/suits/jobs.dm:196]:
```suggestion
	to_chat(usr, SPAN_NOTICE("You [opened ? "unzip" : "zip"] \the [src]."))
```

File: code/modules/clothing/suits/storage.dm [hunk 2]
```diff
@@ -37,29 +37,31 @@
...
-		to_chat(usr, "You unbutton \the [src].")
-	else //in case some goofy admin switches icon states around without switching the icon_open or icon_closed
-		to_chat(usr, "You attempt to button-up the velcro on \the [src], before promptly realising how silly you are.")
-		return
-	update_clothing_icon()	//so our overlays update
+	toggle_open()
 
+/obj/item/clothing/suit/storage/toggle/proc/toggle_open()
+	opened = !opened
+	to_chat(usr, "You [opened ? "unbutton" : "button up"] \the [src].")
```
Comment (NonQueueingMatt) [inline code/modules/clothing/suits/storage.dm:54]:
```suggestion
	to_chat(usr, SPAN_NOTICE("You [opened ? "unbutton" : "button up"] \the [src]."))
```

File: code/modules/customitems/item_defines.dm [hunk 3]
```diff
@@ -2221,56 +2199,44 @@ All custom items with worn sprites must follow the contained sprite system: http
 		graduation.</font> - <font face='Times New Roman'>Teremun A. M.</font></i></small>"))
 		return
 
+/obj/item/clothing/suit/storage/toggle/labcoat/fluff/mekesatis_labcoat/toggle_open()
+	if(!changing)
+		opened = !opened
+		to_chat(usr, "You [opened ? "unbutton" : "button up"] \the [src].")
```
Comment (NonQueueingMatt) [inline code/modules/customitems/item_defines.dm:2205]:
```suggestion
		to_chat(usr, SPAN_NOTICE("You [opened ? "unbutton" : "button up"] \the [src]."))
```

File: code/modules/clothing/suits/jobs.dm (hunk 1 above)
Comment (NonQueueingMatt) [reply code/modules/clothing/suits/jobs.dm:196]: why isn't this a subtype of suit/storage/toggle actually?

File: code/modules/clothing/head/jobs.dm [hunk 4]
```diff
@@ -199,29 +199,3 @@
 /obj/item/clothing/head/det/grey
 	icon_state = "grey_fedora"
 	desc = "A grey fedora - either the cornerstone of a detective's style or a poor attempt at looking cool, depending on the person wearing it."
-
-/obj/item/clothing/head/det/technicolor
```
Comment (NonQueueingMatt) [inline code/modules/clothing/head/jobs.dm:203]: these removals should probably be documented

File: code/game/objects/items/weapons/paint.dm [hunk 5]
```diff
@@ -1,82 +0,0 @@
-//NEVER USE THIS IT SUX	-PETETHEGOAT
-//THE GOAT WAS RIGHT - RKF
-
-/obj/item/reagent_containers/glass/paint
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/paint.dm:4]: document this removal and tell me why exactly this is removed

Comment (alsoandanswer) [reply code/game/objects/items/weapons/paint.dm:4]: 1. They aren't mapped in at all, to my knowledge.

2. Their only purpose was to recolor the detective trenchcoats. Which even then, was terrible, hardcoded dmi nonsense

3. Not referenced anywhere else in code other than here.

File: code/modules/clothing/suits/jobs.dm (hunk 1 above)
Comment (alsoandanswer) [reply code/modules/clothing/suits/jobs.dm:196]: the difference is zipping and the sounds. can't figure out verbs for the life of me, though, and i don't plan on doing repathing of all the hazard vests in this PR.

**Review by fernerr — CHANGES_REQUESTED** [review]
Paint cans shouldn't be removed, it's ordered from cargo from time to time. Its uses go beyond the old technicolor trenchcoat.

**Review by fernerr — COMMENTED** [review]
Goes invisible when toggled:
suit/storage/toggle/det_trench/fluff/leo_coat
suit/storage/toggle/det_jacket

File: code/game/objects/items/weapons/paint.dm [hunk 6]
```diff
@@ -8,7 +8,7 @@
 	icon_state = "paint_empty"
 	item_state = "paintcan"
 	matter = list(DEFAULT_WALL_MATERIAL = 200)
-	w_class = ITEMSIZE_NORMAL
+	w_class = 3.0
```
Comment (fernerr) [inline code/game/objects/items/weapons/paint.dm:11]:
```suggestion
	w_class = ITEMSIZE_NORMAL
```

**Review by fernerr — APPROVED** [review]

**Pull request merged**